lib/modules/manager/sbt/update.ts

```diff
@@ -31,3 +34,31 @@ export function bumpPackageVersion(
 
   return { bumpedContent };
 }
+
+export function updateDependency({
```

Comment:
and also set packageFile to the file, where the replacement should be done

Reply:
But that wouldn't cover the case where the same version is present in the same file right?

Reply:
why it doesn't cover? i don't understand the issue.

renovate will use the `packageFile` later to auto replace the version in that file. if multiple deps are referring to the same variable, group them, so it's a single PR for a single version.

@rarkins i think it's time for an `srcFile` property on the dependency, so auto replace can use that optional fileName to replace the version in the right place.that would probably also help the other managers (maven, gradle) which currently use a custom update function because of that.

Reply:
I tested
```
 INFO: Failed to parse branchContent - rebasing (repository=username/dummpyrepo, packageFile=main/build.sbt, branch=renovate/com.netflix.archaius-archaius-core-0.x)
       "manager": "sbt"
DEBUG: Rebasing branch after deps list has changed (repository=username/dummpyrepo, packageFile=main/build.sbt, branch=renovate/com.netflix.archaius-archaius-core-0.x)
       "depName": "com.netflix.archaius:archaius-core"
DEBUG: manager.getUpdatedPackageFiles() reuseExistingBranch=false (repository=username/dummpyrepo, branch=renovate/com.netflix.archaius-archaius-core-0.x)
DEBUG: Starting search at index 8024 (repository=username/dummpyrepo, packageFile=main/build.sbt, branch=renovate/com.netflix.archaius-archaius-core-0.x)
       "depName": "com.netflix.archaius:archaius-core"
DEBUG: Found match at index 8024 (repository=username/dummpyrepo, packageFile=main/build.sbt, branch=renovate/com.netflix.archaius-archaius-core-0.x)
       "depName": "com.netflix.archaius:archaius-core"
DEBUG: Could not extract main/build.sbt (repository=username/dummpyrepo, branch=renovate/com.netflix.archaius-archaius-core-0.x)
 WARN: Error updating branch: update failure
```

`doAutoReplace` function does not work since it calls `confirmIfDepUpdated` which refers to `extractPackageFile` function which I replaced by `extractAllPackageFiles`

Reply:
Then it would be great to have `extractPackageFile` back. 

`extractAllPackageFiles` typically calls `extractPackageFile`, see npm or batect mangers. `extractAllPackageFiles` should do the same but on `n` packageFiles, instead of just 1 as in `extractPackageFile`.